src/streams.cpp

```diff
@@ -99,3 +100,18 @@ void AutoFile::write(Span<const std::byte> src)
         }
     }
 }
+
+bool AutoFile::Commit()
+{
+    return ::FileCommit(m_file);
+}
+
+bool AutoFile::IsError()
+{
+    return ferror(m_file);
+}
+
+bool AutoFile::Truncate(unsigned size)
+{
+    return ::TruncateFile(m_file, size);
```

Comment:
Note: It is okay to do this without worrying about `m_position` because `ftruncate` does not modify the position of the file seek pointer. (https://pubs.opengroup.org/onlinepubs/007904975/functions/ftruncate.html)